1click-reconfig.sh

```diff
@@ -0,0 +1,230 @@
+}
+# Regenerate node_key.json
+RegenerateNodeKeyJSON()
+{
+    echo_s "Generate and replace node_key in $NODE_KEY_PATH\n"
+    rm -f $CM_HOME/config/node_key.json
+    $CM_BINARY tendermint show-node-id --home $CM_HOME > /dev/null 2>&1
+    ShowNodeKeyInfo
+}
+# print node_key and node_id
+ShowNodeKeyInfo()
+{
+    NODE_ID=$($CM_BINARY tendermint show-node-id --home $CM_HOME)
+    NODE_KEY=$(cat $NODE_KEY_PATH | jq .priv_key.value -r)
+    echo_s "You may want to save node_id and node_key for later use\n"
+    echo_s "node_id: \033[32m$NODE_ID\033[0m\n"
+    echo_s "node_key: \033[32m$NODE_KEY\033[0m\n"
+}
+# allow gossip this ip
+AllowGossip()
+{
+    # find IP
+    IP=$(curl -s http://checkip.amazonaws.com)
+    if [[ -z "$IP" ]] ; then
+        read -p 'What is the public IP of this server?: ' IP
+    fi
+    echo_s "✅ Added public IP to external_address in chain-maind config.toml for p2p gossip\n"
+    sed -i "s/^\(external_address\s*=\s*\).*\$/\1\"$IP:26656\"/" $CM_CONFIG
+}
+EnableStateSync()
+{
+    RPC_SERVERS=$(curl -sS $NETWORK_JSON | jq -r ".\"$NETWORK\".endpoint.rpc")
+    LASTEST_HEIGHT=$(curl -s $RPC_SERVERS/block | jq -r .result.block.header.height)
+    BLOCK_HEIGHT=$((LASTEST_HEIGHT - 300))
+    TRUST_HASH=$(curl -s "$RPC_SERVERS/block?height=$BLOCK_HEIGHT" | jq -r .result.block_id.hash)
+    sed -i "s/^\(seeds\s*=\s*\).*\$/\1\"\"/" $CM_CONFIG
+    sed -i "s/^\(persistent_peers\s*=\s*\).*\$/\1\"$SEEDS\"/" $CM_CONFIG
+    sed -i "s/^\(trust_height\s*=\s*\).*\$/\1$BLOCK_HEIGHT/" $CM_CONFIG
+    sed -i "s/^\(trust_hash\s*=\s*\).*\$/\1\"$TRUST_HASH\"/" $CM_CONFIG
+    sed -i "s/^\(enable\s*=\s*\).*\$/\1true/" $CM_CONFIG
+    sed -i "s|^\(rpc_servers\s*=\s*\).*\$|\1\"$RPC_SERVERS,$RPC_SERVERS\"|" $CM_CONFIG
+}
+DisableStateSync()
+{
+    sed -i "s/^\(enable\s*=\s*\).*\$/\1false/" $CM_CONFIG
+}
+shopt -s extglob
+checkout_network()
+{
+    mapfile -t arr < <(curl -sS $NETWORK_JSON | jq -r 'keys[]')
+    echo_s "You can select the following networks to join"
+    for i in "${!arr[@]}"; do
+        printf '\t%s. %s\n' "$i" "${arr[i]}"
+    done
+
+    read -p "Please choose the network to join by index (0/1/...): " index
+    case $index in
+        +([0-9]))
+            if [[ $index -gt $((${#arr[@]} - 1)) ]]; then
+                echo_s "Larger than the max index"
+                exit 1
+            fi
+            NETWORK=${arr[index]}
+            echo_s "The selected network is $NETWORK"
+            GENESIS_TARGET_SHA256=$(curl -sS $NETWORK_JSON | jq -r ".\"$NETWORK\".genesis_sha256sum")
+            if [[ ! -f "$CM_GENESIS" ]] || (! echo "$GENESIS_TARGET_SHA256 $CM_GENESIS" | sha256sum -c --status --quiet - > /dev/null 2>&1) ; then
+                echo_s "The genesis does not exit or the sha256sum does not match the target one. Download the target genesis from github."
+                download_genesis
+            fi
+            read -p "Do you want to enable state-sync? (Y/N): " yn
```

Comment:
I am considering to add some small notes/references for state-sync enabling, e.g. give suggestion on "N" if they want to build a full node. What do you think @allthatjazzleo ?